cylc/flow/workflow_files.py

```diff
@@ -1528,8 +1546,11 @@ def install_workflow(
             Another workflow already has this name (unless --redirect).
             Trying to install a workflow that is nested inside of another.
     """
-    if source.name == WorkflowFiles.FLOW_FILE:
-        source = source.parent
+    if not source:
+        source = Path.cwd()
+    else:
+        abort_if_flow_file_in_path(source)
+    source = Path(expand_path(source))
```

Comment:
I stripped out most of this logic in #4823 
```suggestion
    abort_if_flow_file_in_path(source)
```